proto/uber/cadence/api/v1/service_workflow.proto

```diff
@@ -90,6 +93,11 @@ service WorkflowAPI {
   rpc RefreshWorkflowTasks(RefreshWorkflowTasksRequest) returns(RefreshWorkflowTasksResponse);
 }
 
+message RestartWorkflowExecutionRequest {
```

Comment:
if the semantics we're going for are always "current run is terminated", we probably need to support either request-ID or an "only if current" / "only if current == UUID" argument.

otherwise, with a successful-but-timed-out request, you risk restarting a *second* time, and not having a good way to prevent that.

Reply:
what does this mean? "with a successful-but-timed-out request, you risk restarting a second time, and not having a good way to prevent that."

Right now it calls startworkflow with the policy terminate if the wfid has something running.